sdk/core/azure-core/azure/core/polling/_poller.py

```diff
@@ -263,6 +262,7 @@ def wait(self, timeout: Optional[float] = None) -> None:
         """
         if self._thread is None:
             return
+        self._thread.start()            
```

Comment:
We need to also make sure that the thread is `start`ed if someone calls `add_done_callback` too, right?